cmd/kubeadm/app/cmd/completion_test.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"github.com/spf13/cobra"
+	"testing"
+)
+
+func TestNewCmdCompletion(t *testing.T) {
+	var out bytes.Buffer
+	shells := GetSupportedShells()
+	if len(shells) == 0 {
+		t.Errorf("GetSupportedShells() returned zero")
+	}
+	// test NewCmdCompletion with a valid shell.
+	// use a dummy parent command as NewCmdCompletion needs it.
+	parentCmd := &cobra.Command{}
+	args := []string{"completion", shells[0]}
```

Comment:
Iterate over all the supported shells?

Reply:
the only purpose of `TestNewCmdCompletion` is to test the command execution with a valid argument / shell.
all valid shells and error cases are iterated in the next test `TestRunCompletion`.